sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -415,6 +416,10 @@ class ResolveSessionCatalog(
     case ShowColumnsStatement(tbl, ns) =>
       val sql = "SHOW COLUMNS"
       val v1TableName = parseV1Table(tbl, sql).asTableIdentifier
+      if (ns.isDefined && ns.get.length > 1) {
```

Comment:
this check should be done earlier, otherwise it's non-sense to check if the database names are specified consistently.